### ISSUE TITLE
[Mailer] Unable to send templated email

### ISSUE THREAD

**Issue by Flower7C3** [symfony/symfony#35990]
**Symfony mailer version affected**: 5.0.5

**Description**
I am unable to send templated email due the error _A message must have a text or an HTML part or attachments._ I was looking for solution in [Mailer Component documentation](https://symfony.com/doc/current/components/mailer.html) and in [Symfony Framework Mailer documentation.](https://symfony.com/doc/current/mailer.html), but there are no info.

**How to reproduce**

```
$mailer_dsn = $container->getParameter('mailbox_dsn');
$transport = \Symfony\Component\Mailer\Transpor::fromDsn($mailer_dsn);
$mailer = new \Symfony\Component\Mailer\Mailer($transport);
$emailMessage = (new Email())
  ->addTo($to);
  ->subject('Lorem ipsum')
  ->htmlTemplate('email/test.html.twig')
  ->textTemplate('email/test.txt.twig');
$mailer->send($emailMessage);
```

**Possible Solution**

Render message with BodyRenderer before send
```
$bodyRenderer = new \Symfony\Bridge\Twig\Mime\BodyRenderer($container->get('twig'), $params);
$bodyRenderer->render($emailMessage);
```

**Comment by stof** [comment]
Well, if you instantiate your mail yourselves and you want the Twig integration, you need to wire it up as well.
The Mailer component has a MessageListener which is responsible for calling the BodyRenderer. But here, you create your own Mailer without injecting an EventDispatcher in it (so no events will be dispatched) and without configuring a MessageListener in this event dispatcher.

If you are using the Symfony framework, use the mailer service configured by FrameworkBundle (which already wires the MessageListener) rather than doing the instantiation yourselves.

**Comment by Flower7C3** [comment]
I am using Symfony MicroKernel and I am trying to implement everything on
my own.
I think that this note should be in documentation. Could You do that?

**Comment by xabbuh** [comment]
I am closing here as there is nothing to fix in the code. If you feel like the documentation can be clarified somehow, I suggest to create a new issue or a pull request with a change suggestion to the [documentation repository](https://github.com/symfony/symfony-docs).

**Comment by fvtorres** [comment]
Hello, Im trying to use Symfony Mailer With Twig inside Slim 4 but Im running into the same issue.

But I didn`t figure out how to make this work...

I`m receiving the same error:

Type: Symfony\Component\Mime\Exception\LogicException
Code: 0
Message: A message must have a text or an HTML part or attachments.
File: /vendor/symfony/mime/Email.php
Line: 405

**Comment by seanopa** [comment]
The reason you're getting this error "A message must have a text or an HTML part or attachments" is because your environment is incorrect. You should use it as below. Notice FileSystemLoader vs ArrayLoader

`$email = (new TemplatedEmail())
                ->from('no-reply@domain.com')
                ->to(new Address('someone@example.com'))
                ->subject($this->subject)
                ->htmlTemplate('email.twig')
                ->context([]);

 $loader = new FilesystemLoader('/path/to/twig/templates/');

$twigEnv = new Environment($loader);

$twigBodyRenderer = new BodyRenderer($twigEnv);

$twigBodyRenderer->render($email);

$mailer->send($email);`

**Comment by YannisThalgott** [comment]
Une fois c/c puis les import class effectués j'ai ce message d'erreur : **The "/path/to/twig/templates/" directory does not exist ("/path/to/twig/templates/").**

![Screen de l'erreur](https://user-images.githubusercontent.com/96988793/175057818-d38e08ee-7519-45ba-96b1-3be130925202.JPG)

**Comment by stof** [comment]
Well, the FilesystemLoader requires that the directory you give it (where it will look for templates) is a valid directory.

**Comment by krstns** [comment]
How would the Twig handling work with current versions of Symfony? FileSystemLoader is deprecated already. in my case each user can have different SMTP settings so this solution looks like a good fit.

**Comment by stof** [comment]
The Twig FilesystemLoader is not deprecated at all.

In the context of Symfony, you should not need to instantiate Twig or its loader yourselves. TwigBundle manages all the config of the Environment in such case.

**Comment by krstns** [comment]
@stof Thanks for taking time to reply.
Given the context of the discussion, not using the twigBodyRenderer render method results in an empty email.
So I think I need to do this after all (as Taurai mentioned 2 years ago).

Either way, I was using the wrong FilesystemLoader, after fixing this it's fine :)

**Comment by adnen-chouibi** [comment]
Hi there

I tried Seanopa's solution, and it works for me. However, I'm facing an issue with the filter | trans, and I suspect other filters might not work either. I've implemented a workaround by adding this code:

```
$twigEnv->addFilter(new TwigFilter("trans", function ($string) {
    return $this->translator->trans($string);
}));
```


It seems to do the job, but I'm not sure if it's the best solution. Any suggestions for handling dynamic DSN without all the extra code? Thanks!
